src/jobs/models/jobExperimentsHandler.js

```diff
@@ -0,0 +1,63 @@
+
+const chaosExperimentsDbConnector = require('../../chaos-experiments/models/database/databaseConnector'),
+    { v4: uuid } = require('uuid'),
+    logger = require('../../common/logger');
+
+const SEC_TO_MS = 1000;
+const MIN_TO_MS = 60 * 1000;
+const HOUR_TO_MS = 60 * 1000;
+const DAY_TO_MS = 60 * 1000;
+const jobExperimentsIdToTimeout = new Map();
+
+async function setChaosExperimentsIfExist(jobId, jobExperiments) {
+    if (!jobExperiments) {
+        return;
+    }
+    try {
+        const baseTimestamp = Date.now();
+        const experimentIds = jobExperiments.map(experiment => experiment.experiment_id);
+        const experiments = await chaosExperimentsDbConnector.getChaosExperimentsByIds(experimentIds);
+        for (const experimentRequest of jobExperiments) {
+            try {
+                const experiment = experiments.find(e => e.id === experimentRequest.experiment_id);
```

Comment:
maybe  just for readability would be good to separate the content inside the for loop to another function
setSingleJobExperiment - don't like the catch inside the cate

Reply:
I did it and changed the logic to be async (with promise all and map) - plz take a look again

Reply:
🍻